src/components/DayPickerRangeController.jsx

```diff
@@ -600,34 +600,42 @@ export default class DayPickerRangeController extends React.Component {
     if (!day || !isDayVisible(day, currentMonth, numberOfMonths, enableOutsideDays)) {
       return updatedDays;
     }
-    let monthIso = toISOMonthString(day);
-    let month = updatedDays[monthIso] || visibleDays[monthIso];
     const iso = toISODateString(day);
 
+    let updatedDaysAfterAddition = { ...updatedDays };
     if (enableOutsideDays) {
```

Comment:
@ljharb This first change is in the `addModifier` method. The basic premise is that if outside days are enabled that means that days at the beginning and end of the month will exist as a part of two separate `CalendarMonth` objects. As a result if you click on a day, you may want to add modifiers to more than one month instance in the state. This is what this branch of the conditional accomplishes.